lib/fpm/package/deb.rb

```diff
@@ -517,14 +517,23 @@ def output(output_path)
 
     attributes[:deb_systemd] = []
     attributes.fetch(:deb_systemd_list, []).each do |systemd|
-      name = File.basename(systemd, ".service")
-      dest_systemd = staging_path("lib/systemd/system/#{name}.service")
+      name = File.basename(systemd)
+      extname = File.extname(name)
+
+      name_with_extension = if extname.empty?
+                              "#{name}.service"
+                            elsif [".service", ".timer"].include?(extname)
+                              name
+                            else
+                              raise ArgumentError, "Invalid systemd unit file extension: #{extname}. Expected .service or .timer, or no extension."
```

Comment:
This should raise FPM::InvalidPackageConfiguration so that fpm prints an error message. With ArgumentError, fpm will exit and print a stack trace.